### PR TITLE
feat: ✨ mod directory creation

### PR DESCRIPTION
**Pull request by KANAjetzt**
Completed all the basics in the `add_mod()` function in *create_mod.gd*. 

- [x]  Validate Mod-ID
- [x]  Add Mod Folder to *mods-unpacked*
    - [x]  Check if *mods-unpacked* dir exists
        - [x]  If not create it
    - [x]  Check if mod dir with mod-id exists
        - [x]  If not create it
            - [x]  Create *Author-ModName* dir
        - [x]  If so show error message ~and ask if to connect instead~
    - [x]  Add *mod_main.gd*
    - [x]  Add *manifest.json*
- [x]  Update the Export Settings UI

Also added a template directory with the example *mod_main.gd* and *manifest.json*

### CODE REVIEW

File: addons/mod_tool/utils.gd [hunk 1]
```diff
@@ -55,6 +56,38 @@ static func output_error(message: String) -> void:
 	ModToolStore.label_output.append_bbcode("\n [color=%s]ERROR: %s[/color]" % [ModToolStore.error_color, message])
 
 
+static func output_info(message: String) -> void:
+	ModToolStore.label_output.append_bbcode("\n" + message)
+
+
+# Validates the mod directory name based on the rules defined in the ModLoader ModManifest Class.
+static func validate_mod_dir_name(dir_name: String) -> bool:
```
Comment (Qubus0) [inline addons/mod_tool/utils.gd:64]: this should be the same as `ModManifest.is_mod_id_valid` right?

File: addons/mod_tool/dock.tscn [hunk 2]
```diff
@@ -503,14 +505,69 @@ script = ExtResource( 2 )
...
+
+[node name="HBox" parent="Popup/CreateMod/Settings/Scroll/VBox" index="1"]
+margin_right = 380.0
+
+[node name="Label" parent="Popup/CreateMod/Settings/Scroll/VBox/HBox" index="0"]
+margin_right = 188.0
+
+[node name="ThunderstoreExport" parent="Popup/CreateMod/Settings/Scroll/VBox/HBox" index="1"]
+margin_left = 192.0
+margin_right = 380.0
```
Comment (Qubus0) [inline addons/mod_tool/dock.tscn:546]: I don't think we need that in the create window

File: addons/mod_tool/dock.tscn [hunk 3]
```diff
@@ -503,14 +505,69 @@ script = ExtResource( 2 )
...
+visible = true
+anchor_left = 0.5
+anchor_top = 0.5
+anchor_right = 0.5
+anchor_bottom = 0.5
+margin_left = -200.0
+margin_top = -75.0
+margin_right = 200.0
+margin_bottom = 75.0
+window_title = "Create new mod"
```
Comment (Qubus0) [inline addons/mod_tool/dock.tscn:519]: needs some min_size, otherwise the window becomes tiny and unreadable for me

**Review by Qubus0 — APPROVED** [review]
looks good overall, just a few small things

File: addons/mod_tool/dock.tscn (hunk 2 above)
Comment (KANAjetzt) [reply addons/mod_tool/dock.tscn:546]: Hmm I have this in my notes:

If Thunderstore Export == true

- [ ]  Add *[README.md](http://readme.md/)* to root dir
- [ ]  Add *manifest.json* to root dir
- [ ]  Add *CHANGELOG.md* to root dir
- [ ]  Add *icon.png* to root dir

File: addons/mod_tool/dock.tscn (hunk 3 above)
Comment (KANAjetzt) [reply addons/mod_tool/dock.tscn:519]: Made it resizable and gave it a min size (350, 150) 👍

File: addons/mod_tool/utils.gd (hunk 1 above)
Comment (KANAjetzt) [reply addons/mod_tool/utils.gd:64]: How did I miss the existence of that one 👀 

That would be something along this lines - right?
`ModManifest.is_mod_id_valid(ModToolStore.name_mod_dir, ModToolStore.name_mod_dir, "", true)`

Comment (KANAjetzt) [reply addons/mod_tool/utils.gd]: fixed [d5bacda](https://github.com/GodotModding/godot-mod-tool/pull/24/commits/d5bacda1af36d1b27b60defa6c54ccc20f4c1466) 👍

File: addons/mod_tool/dock.tscn (hunk 2 above)
Comment (KANAjetzt) [reply addons/mod_tool/dock.tscn:550]: We can hide it for now.
*[9c1892b](https://github.com/GodotModding/godot-mod-tool/pull/24/commits/9c1892bc47a7fc01846a6ebb54104e6a74c51adf)*

Comment (Qubus0) [reply addons/mod_tool/dock.tscn:550]: the checkbox in the bottom panel should suffice. It should be just an extra processing step that happens on export, not for the mod structure. We can just add those files to the mod directory, like our manifest and move all the files needed by thunderstore one level out before zipping. That's why I placed the checkbox in the export dialog, not with mod creation

Comment (KANAjetzt) [reply addons/mod_tool/dock.tscn]: We decided to always add the thunderstore files to mod directory
*[228404f](https://github.com/GodotModding/godot-mod-tool/pull/24/commits/228404f3ac2630b7d3c1f9fd97e01fc77f3aec6e)*

**Pull request merged**